toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt

```diff
@@ -109,6 +111,42 @@ public fun MapView(
             }
         }
     }
+
+    GraphicsOverlaysUpdater(graphicsOverlays, mapView)
+}
+
+/**
+ * Update the [mapView]'s graphicsOverlays property to reflect changes made to the
+ * [graphicsOverlayCollection] based on the type of [GraphicsOverlayCollection.ChangedEvent]
+ */
+@Composable
+private fun GraphicsOverlaysUpdater(
+    graphicsOverlayCollection: GraphicsOverlayCollection,
+    mapView: MapView
+) {
+    LaunchedEffect(graphicsOverlayCollection) {
+        // sync up the MapView with the new graphics overlays
+        mapView.graphicsOverlays.clear()
+        graphicsOverlayCollection.forEach {
+            mapView.graphicsOverlays.add(it)
+        }
+        // start observing graphicsOverlays for subsequent changes
+        graphicsOverlayCollection.changed.collect { changedEvent ->
+            when (changedEvent) {
+                // On GraphicsOverlay added:
+                is GraphicsOverlayCollection.ChangedEvent.Added ->
+                    changedEvent.element?.let { mapView.graphicsOverlays.add(it) }
```

Comment:
I think the null check is redundant now:
```suggestion
                    mapView.graphicsOverlays.add(changedEvent.element)
```